pyresample/test/test_gradient.py

```diff
@@ -249,25 +249,29 @@ def test_resample_area_to_area_3d_single_channel(self):
         assert res.shape == (1, ) + self.dst_area.shape
         assert np.allclose(res[0, :, :], 1.0)
 
-    def test_resample_swath_to_area_2d(self):
+    @pytest.mark.parametrize("input_dtype", (np.float32, np.float64))
+    def test_resample_swath_to_area_2d(self, input_dtype):
         """Resample swath to area, 2d."""
-        data = xr.DataArray(da.ones(self.src_swath.shape, dtype=np.float64),
+        data = xr.DataArray(da.ones(self.src_swath.shape, dtype=input_dtype),
                             dims=['y', 'x'])
         with np.errstate(invalid="ignore"):  # 'inf' space pixels cause runtime warnings
             res = self.swath_resampler.compute(
                 data, method='bil').compute(scheduler='single-threaded')
+        assert res.dtype == data.dtype
```

Comment:
Sorry for this small but annoying request, but could you get the dask-based result and compute the numpy result from it and compare their dtypes? It is the fastest way to catch dask bugs where dask thinks it is one dtype but the final computed array is another.

Reply:
Very good idea indeed! Let's see what I can come up with.

Reply:
I used `values` in the end, tell me what you think.

Reply:
You *could* even assert that the returned array type is dask.

Reply:
> Correct me if I'm wrong, but the above `res = ` line includes a `.compute(scheduler='single-threaded')` so the result of that is already dask-computed. I think you'd need to split that line into two steps: one for the dask array based result, then the computed result (numpy), then do the `res_np` and `res_dask` comparison (or whatever you end up naming the two variables).

You right of course, sorry about that. Should be fixed now. I'm also making sure the shape is right both with numpy and dask arrays.

Reply:
> You _could_ even assert that the returned array type is dask.

In the spirit of duck typing, I will avoid that.

Reply:
Yeah, but this is the tests so I disagree. You are passing in a specific type, you expect a specific type back. If you got a numpy array when you provided a dask array as input then that would be a big bug. Not to mention you could/should check for how many dask computations are happening during the processing and make sure it is zero (or whatever you expect).

Duck typing makes sense in the work code, not the test code...in my opinion.